src/main/java/com/mraof/minestuck/item/crafting/alchemy/GristHelper.java

```diff
@@ -138,33 +143,67 @@ public static void increase(World world, PlayerIdentifier player, GristSet set)
 		newCache.addGrist(set);
 		data.setGristCache(newCache);
 	}
-	
-	private static void notify(MinecraftServer server, PlayerIdentifier player, ITextComponent type, long difference, String action)
+
+	public static void notify(MinecraftServer server, PlayerIdentifier player, GristSet set, boolean increase)
 	{
-		if(MinestuckConfig.showGristChanges.get())
+		if (gristMessages.get())
 		{
-			if (player != null)
+			Map<GristType, Long> reqs = set.getMap();
+			for (Entry<GristType, Long> pairs : reqs.entrySet())
 			{
-				ServerPlayerEntity client = player.getPlayer(server);
-				if(client != null)
+				ITextComponent type = pairs.getKey().getDisplayName();
+				long difference = pairs.getValue();
+				if (increase)
+				{
+					sendGristMessage(server, player, new TranslationTextComponent("You gained %s %s grist.", difference, type));
+				}
+				else
+				{
+					sendGristMessage(server, player, new TranslationTextComponent("You lost %s %s grist.", difference, type));
+				}
+			}
+		}
+	}
+	public static void notifyEditPlayer(MinecraftServer server, PlayerIdentifier player, GristSet set, boolean increase) {
+		if(gristMessagesEditMode.get())
+		{
+			SburbConnection sc = SkaianetHandler.get(server).getActiveConnection(player);
+			if (sc == null)
+				return;
+
+			EditData ed = ServerEditHandler.getData(server, sc);
+			if (ed == null)
+				return;
+
+			Map<GristType, Long> reqs = set.getMap();
+			for (Entry<GristType, Long> pairs : reqs.entrySet())
+			{
+				ITextComponent type = pairs.getKey().getDisplayName();
+				long difference = pairs.getValue();
+				if (increase)
 				{
-					//"true" sends the message to the action bar (like bed messages), while "false" sends it to the chat.
-					client.sendStatusMessage(new TranslationTextComponent("You " + action + " " + difference + " " + type + " grist."), true);//TODO Translation
+					sendGristMessage(server, IdentifierHandler.encode(ed.getEditor()), new TranslationTextComponent("You have refunded %s of %s's %s grist.", difference, player.getUsername(), type));
+				} else
+				{
+					sendGristMessage(server, IdentifierHandler.encode(ed.getEditor()), new TranslationTextComponent("You have spent %s of %s's %s grist.", difference, player.getUsername(), type));
 				}
 			}
 		}
 	}
-	
-	private static void notifyEditPlayer(MinecraftServer server, PlayerIdentifier player, ITextComponent type, long difference, String action)
+
+	private static void sendGristMessage(MinecraftServer server, PlayerIdentifier player, ITextComponent message)
 	{
-		SburbConnection sc = SkaianetHandler.get(server).getActiveConnection(player);
-		if(sc == null)
-			return;
-		
-		EditData ed = ServerEditHandler.getData(server, sc);
-		if(ed == null)
-			return;
-		
-		notify(server, IdentifierHandler.encode(ed.getEditor()), type, difference, action);
+		if (MinestuckConfig.showGristChanges.get())
```

Comment:
Do this config check in the functions above instead of with those two config options that you added